tests-pending/rpc_pending_test.go

```diff
@@ -0,0 +1,332 @@
+		os.Exit(1)
+	}
+
+	// Start all tests
+	code := m.Run()
+	os.Exit(code)
+}
+
+func TestEth_Pending_GetBalance(t *testing.T) {
+	var res hexutil.Big
+	rpcRes := util.Call(t, "eth_getBalance", []string{addrA, "latest"})
+	err := res.UnmarshalJSON(rpcRes.Result)
+	require.NoError(t, err)
+	preTxLatestBalance := res.ToInt()
+
+	rpcRes = util.Call(t, "eth_getBalance", []string{addrA, "pending"})
+	err = res.UnmarshalJSON(rpcRes.Result)
+	require.NoError(t, err)
+	preTxPendingBalance := res.ToInt()
+
+	t.Logf("Got pending balance %s for %s pre tx\n", preTxPendingBalance, addrA)
+	t.Logf("Got latest balance %s for %s pre tx\n", preTxLatestBalance, addrA)
+
+	param := make([]map[string]string, 1)
+	param[0] = make(map[string]string)
+	param[0]["from"] = "0x" + fmt.Sprintf("%x", from)
+	param[0]["to"] = addrA
+	param[0]["value"] = "0xA"
+	param[0]["gasLimit"] = "0x5208"
+	param[0]["gasPrice"] = "0x1"
+
+	_ = util.Call(t, "eth_sendTransaction", param)
+
+	rpcRes = util.Call(t, "eth_getBalance", []string{addrA, "pending"})
+	err = res.UnmarshalJSON(rpcRes.Result)
+	require.NoError(t, err)
+	postTxPendingBalance := res.ToInt()
+	t.Logf("Got pending balance %s for %s post tx\n", postTxPendingBalance, addrA)
+
+	require.Equal(t, preTxPendingBalance.Add(preTxPendingBalance, big.NewInt(10)), postTxPendingBalance)
+
+	rpcRes = util.Call(t, "eth_getBalance", []string{addrA, "latest"})
+	err = res.UnmarshalJSON(rpcRes.Result)
+	require.NoError(t, err)
+	postTxLatestBalance := res.ToInt()
+	t.Logf("Got latest balance %s for %s post tx\n", postTxLatestBalance, addrA)
+
+	require.Equal(t, preTxLatestBalance, postTxLatestBalance)
+	// 1 if x > y; where x is postTxPendingBalance, y is preTxPendingBalance
+	if postTxLatestBalance.Cmp(preTxPendingBalance) != 0 {
+		t.Errorf("expected balance: %d, got: %s", 0, res.String())
+	}
+}
+
+func TestEth_Pending_GetTransactionCount(t *testing.T) {
+	currentNonce := util.GetNonce(t, "latest")
+	t.Logf("Current nonce is %d", currentNonce)
+
+	param := make([]map[string]string, 1)
+	param[0] = make(map[string]string)
+	param[0]["from"] = "0x" + fmt.Sprintf("%x", from)
+	param[0]["to"] = addrA
+	param[0]["value"] = "0xA"
+	param[0]["gasLimit"] = "0x5208"
+	param[0]["gasPrice"] = "0x1"
+
+	_ = util.Call(t, "eth_sendTransaction", param)
+
+	pendingNonce := util.GetNonce(t, "pending")
+	latestNonce := util.GetNonce(t, "latest")
+	t.Logf("Latest nonce is %d", latestNonce)
+	require.Equal(t, currentNonce, latestNonce)
+	t.Logf("Pending nonce is %d", pendingNonce)
+	require.NotEqual(t, latestNonce, pendingNonce)
+
+	require.Greater(t, uint64(pendingNonce), uint64(latestNonce))
+}
+
+func TestEth_Pending_GetBlockTransactionCountByNumber(t *testing.T) {
+	rpcRes := util.Call(t, "eth_getBlockTransactionCountByNumber", []interface{}{"pending"})
+	var preTxPendingTxCount hexutil.Uint
+	err := json.Unmarshal(rpcRes.Result, &preTxPendingTxCount)
+	require.NoError(t, err)
+	t.Logf("Pre tx pending nonce is %d", preTxPendingTxCount)
```

Comment:
nonce -> tx count?

Reply:
i dont think those are the same things. the tx count that's in the block is less than that of the acct nonce. so we're trying to check for the block's pendingTx count